libdrgn/python/object.c

```diff
@@ -908,6 +908,30 @@ static PyObject *DrgnObject_str(DrgnObject *self)
 	return ret;
 }
 
+static PyObject *DrgnObject_repr_pretty(DrgnObject *self, PyObject *args,
+					PyObject *kwds)
+{
+	static char *keywords[] = {"p", "cycle", NULL};
+	PyObject *p, *str_obj, *ret;
+	int cycle;
+
+	if (!PyArg_ParseTupleAndKeywords(args, kwds, "Op:_repr_pretty_", keywords,
+					 &p, &cycle))
+		return NULL;
+
+	if (cycle)
+		return PyObject_CallMethod(p, "text", "s", "...");
+
+	str_obj = DrgnObject_str(self);
```

Comment:
If you use `PyObject_Str()` instead of `DrgnObject_str` (and make `self` a `PyObject *`), then you can reuse the same implementation for all of the types that need it.

Reply:
To elaborate on this, you could name this function something more generic (`repr_pretty_from_str()` or something like that?) put it somewhere common (maybe `libdrgn/module/python.c`), and use it in the method table for the other classes.